Cast INTERVAL and UUID to VARCHAR

guenp opened the pull request:
Fixes https://github.com/MotherDuck-Open-Source/duckdb-power-query-connector/issues/7.

Power Query/Power BI currently does not support `INTERVAL` and loading data with this type results in an error:
```
ODBC: ERROR [22007] ODBC_DuckDB->GetDataStmtResult Not implemented Error: Unimplemented type for cast (INTERVAL -> TIME)
```

Similarly, implementing a cast to `TIME` also won't work because in Power Query, valid values for `TIME` hours are 0-23.

`UUID` has a similar problem that it throws an `Unimplemented type for cast (UUID -> DOUBLE)` error.

This PR casts `INTERVAL` and `UUID` to `VARCHAR` by default. The `INTERVAL` value can be parsed separately in Power Query, for instance by using `Duration.FromText` (see [docs](https://learn.microsoft.com/en-us/powerquery-m/duration-functions)).

<img width="671" alt="image" src="https://github.com/duckdb/duckdb/assets/4041805/11873f32-40c3-44c0-a1c1-9e20d7ddbebf">

### Comparing to Postgres

it looks like Postgres doesn't offer interval support for the ODBC driver either:
https://github.com/hiinoue/psqlodbc/blob/f28b1dcc197c8775d6a34a419d73a0cfa97fa643/pgtypes.h#L82

```
/*
 *	SQL_INTERVAL support is disabled because I found
 *	some applications which are unhappy with it.
 *
#define	PG_INTERVAL_AS_SQL_INTERVAL
 */
```

maiadegraaf commented on test/tests/catalog_functions.cpp:
```diff
@@ -76,6 +76,7 @@ void TestGetTypeInfo(HSTMT &hstmt, std::map<SQLSMALLINT, SQLULEN> &types_map) {
 	    SQL_DOUBLE,
 	    SQL_VARCHAR,
 	    SQL_VARBINARY,
+	    SQL_UNKNOWN_TYPE,
```
Can you explain why you've added this?

guenp replied:
This test runs `SQLGetTypeInfo` with `SQL_ALL_TYPES`, which returns all types listed in `ApiInfo::ODBC_SUPPORTED_SQL_TYPES`. Sine we added a line item to that list (`"'UUID'"`) we need to add its corresponding ODBC type here as well (`0` or `SQL_UNKNOWN_TYPE`).

maiadegraaf reviewed (changes requested):
Thanks for the PR! 
Since Postgres doesn't support intervals either, I don't see a problem with casting them to `VARCHAR`.  Nonetheless, I do wonder if there any applications that do support `SQL_INTERVAL`, and how would this change affect them?

Could you also add a test for `UUID` in `test/tests/col_attribute`. And potentially one for an `INTERVAL` type as well.

I also left one small comment below.

guenp commented:
@maiadegraaf thanks for your comments! I've added a test for `UUID`, it looks like there was already a test for `INTERVAL`.

It looks like the `sql_desc_type` is updated by the `ApiInfo::ODBC_SUPPORTED_SQL_TYPES`, but `sql_desc_type_name` is not, instead, it uses `InternalType()` which returns `INT128` for `UUID` by default. I've added a line to add this in as a patch but this might need to be fixed elsewhere.

As for how this will work with applications that support `INTERVAL`, I can't find any examples so am not sure, but I imagine it will depend on the application. Given the behavior in Postgres I assume this is a rare scenario.

maiadegraaf reviewed (approved):
Great! Thanks for adding the tests! LGTM :)
